### PR TITLE
Vehicle improvements

### PR DESCRIPTION
**Pull request by deepvelocity**
#### Summary
Features "Vehicle improvements: togglable pre-collision system and foldable turrets"

#### Purpose of change

1. Autopilot self-driving features can behave in a weird manner due to pre-collision check. Example: vehicle will refuse to go up the ramp on a bridge (viewing ramp as a collision candidate). It also prevents the player from building blade-covered mowing bots, kamikaze bots and probably other fun stuff.
2. We have foldable robot driving unit and turret mount, but not a single foldable mounted gun. Also, unfoldable autopilot. This PR addresses this problem.

#### Describe the solution
1. Separate the pre-collision check into a separate method, introduce a separate variable to enable or disable pre-collision system. This way, existing code base will not suffer changes and the player will have more room for customization (finally "programmable" part of the autopilot will make sense!)
2. Add option to enable/disable this feature in autopilot menu.
3. Make every Pistol and SMG mounted gun foldable using the gun's item volume.

#### Describe alternatives you've considered
The only concern I have is around the definition of foldable. While it appears that generally Cataclysm DDA refers to things as foldable when they can actually be folded, it also doesn't seem logical that the player cannot pick up a single tile vehicle using its full volume for the final item volume. Making only SMGs and Pistols foldable seems like a good middle ground.

#### Testing
Local build with tests enabled passed. All features (togglable CAS system, foldable turrets/autopilot, remote control selection skip, sound beeps) work as intended.

Also tested with animal-powered vehicle to confirm that those do not produce the same sound.

### CODE REVIEW

**Comment by Qrox** [comment]
Why closed?

**Comment by deepvelocity** [comment]
> Why closed?

This PR contained changes from another PR I raised, which had not been merged yet at the moment. Re-opened PR.

File: data/json/vehicleparts/vehicle_parts.json [hunk 1]
```diff
@@ -468,6 +468,7 @@
     "broken_color": "red",
     "damage_modifier": 80,
     "durability": 100,
+    "folded_volume": "2500 ml",
```
Comment (Maleclypse) [inline data/json/vehicleparts/vehicle_parts.json:471]:
```suggestion
    "folded_volume": "2500 ml"
```

Comment (Maleclypse) [reply data/json/vehicleparts/vehicle_parts.json:471]: Resetting test suite

File: data/json/vehicleparts/vehicle_parts.json [hunk 2]
```diff
@@ -468,6 +468,7 @@
     "broken_color": "red",
     "damage_modifier": 80,
     "durability": 100,
+    "folded_volume": "2500 ml"
```
Comment (Maleclypse) [inline data/json/vehicleparts/vehicle_parts.json:471]:
```suggestion
    "folded_volume": "2500 ml",
```

**Pull request merged**